src/Symfony/Bundle/FrameworkBundle/Controller/ControllerResolver.php

```diff
@@ -78,7 +78,11 @@ protected function createController($controller)
      */
     protected function instantiateController($class)
     {
-        $controller = parent::instantiateController($class);
+        if ($this->container->has($class)) {
+            $controller = $this->container->get($class);
+        } else {
+            $controller = parent::instantiateController($class);
+        }
 
         if ($controller instanceof ContainerAwareInterface) {
```

Comment:
this should not be done when getting the class from the container. It should happen only when instantiating